src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -400,22 +426,85 @@ long TriangleCountExecutor::getTriangleCount(int graphId, std::string host, int
+                std::set<string>::iterator partitionSetIterator;
                 std::set<string> transferRequireFiles;
                 std::string combinationKey = "";
                 std::string availableFiles = "";
                 std::string transferredFiles = "";
                 bool isAggregateValid = false;
 
+                for (listIterator = fileList.begin(); listIterator != fileList.end(); ++listIterator) {
+                    std::string fileName = *listIterator;
+
+                    size_t lastindex = fileName.find_last_of(".");
+                    string rawFileName = fileName.substr(0, lastindex);
+
+                    std::vector<std::string> fileNameParts = utils.split(rawFileName,'_');
+
+                    for (int index = 2; index < fileNameParts.size(); ++index) {
+                        partitionSet.insert(fileNameParts[index]);
+                    }
+
+                }
+
+                if (partitionSet.find(std::to_string(partitionId)) == partitionSet.end()) {
+                    continue;
+                }
+
+                /*aggregateWeightMutex.lock();
+                triangleCount_logger.log("###COMPOSITE### Aggregator MUTEX Aquired ", "info");
+                for (partitionSetIterator = partitionSet.begin(); partitionSetIterator != partitionSet.end(); ++partitionSetIterator) {
+                    std::string partitionIdString = *partitionSetIterator;
+                    int currentPartitionId = atoi(partitionIdString.c_str());
+                    tempWeightMap[currentPartitionId] = aggregateWeightMap[currentPartitionId];
+                }
+
+                int currentWorkerWeight = tempWeightMap[partitionId];
+                pair<int, int> entryWithMinValue = make_pair(partitionId,currentWorkerWeight);
+
+                map<int, int>::iterator currentEntry;
+
+                for (currentEntry = aggregateWeightMap.begin(); currentEntry != aggregateWeightMap.end(); ++currentEntry) {
+                    if (entryWithMinValue.second > currentEntry->second) {
+                        entryWithMinValue = make_pair(currentEntry->first,currentEntry->second);
+                    }
+                }
+
+                if (entryWithMinValue.first == partitionId) {
+                    int currentWeight = aggregateWeightMap[entryWithMinValue.first];
+                    currentWeight++;
+                    aggregateWeightMap[entryWithMinValue.first] = currentWeight;
+                    triangleCount_logger.log("###COMPOSITE### Aggregator Initiated : Partition ID: " + std::to_string(partitionId) +" Host: " + host + " Port: " + std::to_string(port) + " Weight : " + std::to_string(currentWeight), "info");
+                } else {
+                    continue;
+                }
+
+
+
+                aggregateWeightMutex.unlock();
+                triangleCount_logger.log("###COMPOSITE### Aggregator MUTEX Released ", "info");*/
+
+                if (proceedOrNot(partitionSet,partitionId)) {
+
```

Comment:
We should not keep an empty block here. We should use negation operator in the if condition and remove this empty block.

Reply:
Fixed this issue

Reply:
Ok